core/src/wallet/tezos/explorers/ExternalTezosLikeBlockchainExplorer.cpp

```diff
@@ -83,21 +98,47 @@ namespace ledger {
                         if (!json.IsObject() || !json.HasMember(feesField) ||
                             !json[feesField].IsString()) {
                             throw make_exception(api::ErrorCode::HTTP_ERROR,
-                                                 "Failed to get fees from network, no (or malformed) field \"result\" in response");
+                                                 fmt::format("Failed to get fees from network, no (or malformed) field \"{}\" in response", feesField));
                         }
-                        std::string fees = json[feesField].GetString();
-                        // Sometimes network is sending 0 for fees
-                        if (fees == "0") {
-                            fees = api::TezosConfigurationDefaults::TEZOS_DEFAULT_FEES;
-                        } else if (fees.find('.') != std::string::npos) {
-                            fees = api::BigInt::fromDecimalString(fees, 6, ".")->toString(10);
+
+                        // Return 0 if the block had no transaction at all
+                        if (!json.HasMember(txCountField) || !json[txCountField].IsString()) {
+                            return std::make_shared<BigInt>(0);
+                        }
+
+                        const auto totalTx = api::BigInt::fromIntegerString(json[txCountField].GetString(), 10);
+                        const auto totalFees = api::BigInt::fromDecimalString(json[feesField].GetString(), 6, ".");
+                        std::string fees = api::TezosConfigurationDefaults::TEZOS_DEFAULT_FEES;
+                        if (fees != "0" && totalTx->intValue() != 0) {
+                            fees = totalFees->divide(totalTx)->toString(10);
                         }
-                        // Since nodes are giving some awkward values, we set a threshold to avoid having really fees
-                        // Factor for threshold is inspired from other XTZ wallets
+                        // Since nodes are giving some awkward values, we set a threshold to avoid
+                        // having really fees Factor for threshold is inspired from other XTZ
+                        // wallets
```

Comment:
Need a `.` between `fees` and `Factor`, otherwise it doesn't make any sense :grammarnazi: